src/clock.rs

```diff
@@ -39,7 +38,15 @@ impl Instant {
             (
                 InstantKind::Virtual { nanoseconds },
                 InstantKind::Virtual { nanoseconds: earlier },
-            ) => Duration::from_nanos(nanoseconds.saturating_sub(earlier)),
+            ) => {
+                // If it exceeded u64::MAX nanosecond, we will just keep u64::MAX nanosecond,
+                // Duration can't take in more than u64::MAX.
+                let duration = match u64::try_from(nanoseconds.saturating_sub(earlier)) {
+                    Ok(nanosecond) => Duration::from_nanos(nanosecond),
+                    Err(_err) => Duration::from_nanos(u64::MAX),
+                };
+                Duration::new(duration.as_secs(), duration.subsec_nanos())
```

Comment:
I don't think there's any reason to do this dance through `Duration::from_nanos`. We should be able to just take the nanosecond value div and mod a billion to get the right `Duration::new` arguments.

Reply:
Ah right, I somehow forgot the existence of mod.